src/k2/persistence/logStream/LogStream.cpp

```diff
@@ -0,0 +1,343 @@
+                std::vector<Payload> payloads;
+                for (auto& response: responses){
+                    auto& [status, resp] = response;
+                    if (!status.is2xxOK()){
+                        throw k2::dto::PlogReadError("unable to read a plog");
+                    }
+                    payloads.push_back(std::move(resp));
+                }
+                // read the contents from WAL 
+                return _readContent(std::move(payloads));
+            });
+        });
+        
+    });
+}
+
+seastar::future<std::vector<Payload> >
+LogStream::_readContent(std::vector<Payload> payloads){
+    std::vector<dto::MetadataElement> streamLog;
+
+    //Format: plogId1, offset1, plogId2, offset2... 
+    bool readId = true;
+    String pending_plogId;
+    for (auto& payload:payloads){
+        payload.seek(0);
+        while (payload.getDataRemaining() > 0){ 
+            if (readId){
+                payload.read(pending_plogId);
+                readId = false;
+            }
+            else{
+                readId = true;
+                uint32_t offset;
+                payload.read(offset);
+                dto::MetadataElement log_entry{.name=pending_plogId, .offset=offset};
+                streamLog.push_back(std::move(log_entry));
+            }
+        } 
+        payload.clear();
+    }
+    dto::MetadataElement log_entry{.name=pending_plogId, .offset=0};
+    streamLog.push_back(std::move(log_entry));
+
+    return seastar::do_with(std::move(streamLog), [this] (auto& streamLog){
+        return _client.info(streamLog[streamLog.size()-1].name)
+        .then([&] (auto&& response){
+            auto& [status, resp] = response;
+            if (!status.is2xxOK()) {
+                throw k2::dto::PlogInfoError("unable to obtain the information of a plog");
+            }
+            streamLog[streamLog.size()-1].offset = std::get<0>(resp);
+            std::vector<seastar::future<std::tuple<Status, Payload> > > readFutures;
+            for (auto& log_entry: streamLog){
+                readFutures.push_back(_client.read(log_entry.name, 0, log_entry.offset));
+            }
+            return seastar::when_all_succeed(readFutures.begin(), readFutures.end());
+        })
+        .then([this] (std::vector<std::tuple<Status, Payload> >&& responses){
+            std::vector<Payload> newPayloads;
+            for (auto& response: responses){
+                auto& [status, resp] = response;
+                if (!status.is2xxOK()){
+                    throw k2::dto::PlogReadError("unable to read a plog");
+                }
+                newPayloads.push_back(std::move(resp));
+            }
+            return seastar::make_ready_future<std::vector<Payload> >(std::move(newPayloads));
+        });
+    });
+}
+
+
+seastar::future<> 
+LogStream::write(Payload payload){
+    return _write(std::move(payload), true);
+}
+
+seastar::future<> 
+LogStream::_write(Payload payload, bool writeToWAL){

```

Comment:
rvalue& for the write payload?

Also this WAL makes no sense to me in the context of a LogStream